packages/governance/src/governParam.js

```diff
@@ -0,0 +1,146 @@
+};
+
+/** @type {ValidateParamChangeBallot} */
+const validateParamChangeBallot = details => {
+  assert(
+    details.method === ChoiceMethod.APPROVAL,
+    X`ChoiceMethod must be APPROVAL, not ${details.method}`,
+  );
+  assert(
+    details.electionType === ElectionType.PARAM_CHANGE,
+    X`ElectionType must be PARAM_CHANGE, not ${details.electionType}`,
+  );
+  assert(
+    details.maxChoices === 1,
+    X`maxChoices must be 1, not ${details.maxChoices}`,
+  );
+  assert(
+    details.quorumRule === QuorumRule.MAJORITY,
+    X`QuorumRule must be MAJORITY, not ${details.quorumRule}`,
+  );
+  assert(
+    details.tieOutcome.noChange,
+    X`tieOutcome must be noChange, not ${details.tieOutcome}`,
+  );
+};
+
+/** @type {AssertBallotConcernsQuestion} */
+const assertBallotConcernsQuestion = (paramName, ballotDetails) => {
+  assert(
+    // @ts-ignore typescript isn't sure the question is a paramChangeQuestion
+    // if it isn't, the assertion will fail.
+    ballotDetails.question.paramSpec.parameterName === paramName,
+    X`expected ${paramName} to be included`,
+  );
+};
+
+/** @type {SetupGovernance} */
+const setupGovernance = async (
+  paramManagerAccessor,
+  poserFacet,
+  contractInstance,
+  timer,
+) => {
+  /** @type {WeakSet<Instance>} */
+  const ballotCounters = new WeakSet();
+
+  /** @type {VoteOnParamChange} */
+  const voteOnParamChange = async (
+    paramSpec,
+    proposedValue,
+    ballotCounterInstallation,
+    deadline,
+  ) => {
+    const paramMgr = E(paramManagerAccessor).get(paramSpec);
+    const paramName = paramSpec.parameterName;
+    const param = await E(paramMgr).getParam(paramName);
+    assertType(param.type, proposedValue, paramName);
+    const outcomeOfUpdateP = makePromiseKit();
+
+    const { positive, negative } = makeParamChangePositions(
+      paramSpec,
+      proposedValue,
+    );
+    const question = harden({
+      paramSpec,
+      contract: contractInstance,
+      proposedValue,
+    });
+    const ballotSpec = looksLikeBallotSpec({
+      method: ChoiceMethod.APPROVAL,
+      question,
+      positions: [positive, negative],
+      electionType: ElectionType.PARAM_CHANGE,
+      maxChoices: 1,
+      closingRule: { timer, deadline },
+      quorumRule: QuorumRule.MAJORITY,
+      tieOutcome: negative,
+    });
+
+    const {
+      publicFacet: counterPublicFacet,
+      instance: ballotCounter,
+    } = await E(poserFacet).addQuestion(ballotCounterInstallation, ballotSpec);
+
+    ballotCounters.add(ballotCounter);
+
+    E(counterPublicFacet)
+      .getOutcome()
+      .then(outcome => {
+        if (sameStructure(positive, outcome)) {
+          E(paramMgr)
+            [`update${paramName}`](proposedValue)
+            .then(newValue => outcomeOfUpdateP.resolve(newValue))
+            .catch(e => {
+              outcomeOfUpdateP.reject(e);
+            });
+        }
+      })
+      .catch(e => {
+        outcomeOfUpdateP.reject(e);
+      });
```

Comment:
We we discovered verbally, there's a missing error case which currently would leave the outcomeOfUpdate promise unresolved.

```suggestion
      .then(outcome => {
        if (sameStructure(positive, outcome)) {
          E(paramMgr)
            [`update${paramName}`](proposedValue)
            .then(newValue => outcomeOfUpdateP.resolve(newValue))
            .catch(e => {
              outcomeOfUpdateP.reject(e);
            });
        } else {
          // resolve to indicate a non-erroneous negative outcome
          outcomeOfUpdateP.resolve(something);
        }
      })
      .catch(e => {
        outcomeOfUpdateP.reject(e);
      });
```

Reply:
done